eventstore/store.go

```diff
@@ -294,6 +297,16 @@ func (s *Store) readTxn(m *Model, f func(txn *Txn) error) error {
 
 // Close closes the store
 func (s *Store) Close() {
+	s.lock.Lock()
+	defer s.lock.Unlock()
+	if s.closed {
+		return
+	}
+	s.closed = true
+
+	if s.adapter != nil {
```

Comment:
Use new `Close()` function to wait for the storethread adapter to finish its job, being:
- sending new events from store to threads
- sending new external events from threads to store

This new `Close()` function in this adapter will wait gracefully until we're sure all current things finished correctly. Before this function, lifecycle was handleled by a `context.Context` but that gives not guarantees about finishing gracefully on `ctx.Done()`.